axum-macros/tests/debug_handler/pass/generics_multiple_params_in_arg.rs

```diff
@@ -0,0 +1,49 @@
+use axum::{
+    async_trait,
+    extract::{FromRequest, RequestParts},
+    http::StatusCode,
+    response::IntoResponse,
+};
+use axum_macros::debug_handler;
+
+struct ExampleExtract<T, U> {
+    t: T,
+    u: U,
+}
+
+impl<T, U> Default for ExampleExtract<T, U>
+where
+    T: Default,
+    U: Default,
+{
+    fn default() -> Self {
+        Self {
+            t: Default::default(),
+            u: Default::default(),
+        }
+    }
+}
+
+#[async_trait]
+impl<B, T, U> FromRequest<B> for ExampleExtract<T, U>
+where
+    B: Send,
+    T: Default,
+    U: Default,
+{
+    type Rejection = (StatusCode, &'static str);
+
+    async fn from_request(_req: &mut RequestParts<B>) -> Result<Self, Self::Rejection> {
+        Ok(Default::default())
+    }
+}
+
+#[debug_handler(with(T = String, T = u64, U = i32, U = i16))]
+async fn handler<T, U>(_foo: ExampleExtract<T, U>) -> impl IntoResponse
```

Comment:
I think a separate test showing that a single *invalid* value (i.e. a type that doesn't meet the required constraints) in `with()` generates an error would be very nice.